src/cpu/gemm_convolution.hpp

```diff
@@ -71,7 +75,30 @@ struct gemm_convolution_fwd_t : public primitive_t {
                 bool ok = true;
 
                 for (int i = 0; i < po.len(); i++) {
-                    ok = ok && utils::one_of(po.entry_[i].kind, sum, binary, eltwise, depthwise, quantization);
+                    const auto &post_op = po.entry_[i];
+#if DNNL_X64
+                    using namespace cpu::x64;
+                    cpu_isa_t isa = isa_undef;
+                    if (po.entry_[i].kind == binary) {
+                        auto dst_md = this->dst_md();
+                        if (mayiuse(avx512_core))
+                            isa = avx512_core;
+                        else if (mayiuse(avx2))
+                            isa = avx2;
+                        else if (mayiuse(sse41))
+                            isa = sse41;
+                        if ((isa == isa_undef)
+                                || !binary_injector::is_supported(isa,
+                                        binary_injector::get_src1_desc(
+                                                post_op, *dst_md),
+                                        *dst_md, default_strategies())) {
```

Comment:
@luweizhou2016 , which broadcasting strategies are supported by the `jit_gemm` convolution?

Reply:
> @luweizhou2016 , which broadcasting strategies are supported by the `jit_gemm` convolution?

@maxnick , For jit_gemm_conv itself  it doesn't support any post ops. Forked feature use jit_pp_kernel to handle post_ops.  So jit_pp_kernel invoke jit_uni_binary_post_ops implement in onednn, in theory, jit_uni_binary should have no limitation on broadcast rule after going through the code. Even no broadcast .

For OV side only fuse per-channle or per-tensor for binary post-ops.  Seems @liubo-intel  need to first debug  to check , why post-ops checking  would fail when creating jit_pp kernel. Seems OV only fuse the binary post ops with per OC and per tensor. These should be okay for current check in theory. Thx!

Reply:
Thanks @luweizhou2016 inputs, it seems `broadcasting_strategy_t::no_broadcast` has [higher priority](https://github.com/openvinotoolkit/oneDNN/blob/0f94c0e7b94f64df2b94929279bbeb4f576a6a36/src/common/broadcast_strategy.cpp#L174-L184) than 'broadcasting_strategy_t::per_oc'. so if  OV side only fuse per-channle or per-tensor for binary post-ops. how about use the following way to fix this segmentation-fault issue during `jit_pp_kernel_t ` creation @luweizhou2016 @maxnick :
`--- a/src/cpu/x64/jit_gemm_convolution_utils.cpp
+++ b/src/cpu/x64/jit_gemm_convolution_utils.cpp
@@ -113,13 +113,16 @@ struct jit_pp_kernel_t : pp_kernel_t, public jit_generator {
     static bool post_ops_ok(const convolution_pd_t *pd) {
         const auto& post_ops = pd->attr()->post_ops_;
         auto dst_md = pd->dst_md();
+
+        const bcast_set_t s
+            = {broadcasting_strategy_t::scalar, broadcasting_strategy_t::per_oc};
         for (int i = 0; i < post_ops.len(); i++) {
             const auto &post_op = post_ops.entry_[i];
             if (post_op.is_binary()) {
                 if (!binary_injector::is_supported(isa,
                         binary_injector::get_src1_desc(post_op, *dst_md),
                         *dst_md,
-                        default_strategies())) {
+                        s)) {
                         return false;
                 }
             }`

Reply:
To my understanding, `no_broadcast` is applicable only when both the output and the argument shapes are the same. This may be used by OV in theory (though it's not used now), but at the moment this shouldn't bring any problem. @luweizhou2016 , could you please confirm since you are the author of this feature?

Reply:
Originally the root cause is not in the same shapes, but rather in the fact that in the case of FullyConnected node incorrect bin-post-op argument shape has been sent to the attrs, namely it has wrong rank. But we already have a fix for that issue https://github.com/openvinotoolkit/openvino/pull/23272. The whole idea is to introduce correct binary post op guard checks at the oneDNN level to prevent any unexpected behavior.